helm/external-dns-app/templates/deployment.yaml

```diff
@@ -19,84 +19,99 @@ spec:
-        runAsGroup: {{ .Values.groupID }}
-        fsGroup: {{ .Values.groupID }}
+        runAsUser: {{ .Values.global.securityContext.userID }}
+        runAsGroup: {{ .Values.global.securityContext.groupID }}
+        fsGroup: {{ .Values.global.securityContext.fsGroupID }}
       priorityClassName: giantswarm-critical
       {{- if and (eq .Values.provider "aws") (eq .Values.aws.access "internal") }}
       initContainers:
       - name: wait-for-iam-role
-        image: {{ .Values.image.registry }}/giantswarm/alpine:3.12.0
+        image: {{ .Values.global.image.registry }}/giantswarm/alpine:3.12.0
         command:
         - /bin/sh
         - -c
-        - counter=5; while ! wget -qO- http://169.254.169.254/latest/meta-data/iam/security-credentials/ | grep {{ .Values.clusterID}}-Route53Manager-Role; do echo 'Waiting for iam-role to be available...'; sleep 5; let "counter-=1"  ; if [ "$counter" -eq "0" ]; then exit 1; fi; done
+        - counter=5; while ! wget -qO- http://169.254.169.254/latest/meta-data/iam/security-credentials/ | grep {{ template "aws.iam.role" . }}; do echo 'Waiting for iam-role to be available...'; sleep 5; let "counter-=1"  ; if [ "$counter" -eq "0" ]; then exit 1; fi; done
       {{- end }}
       containers:
-      - name: {{ .Values.name }}
-        image: "{{ .Values.image.registry }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
+      - name: {{ .Release.Name }}
+        image: "{{ .Values.global.image.registry }}/{{ .Values.global.image.name }}:{{ .Values.global.image.tag }}"
         imagePullPolicy: IfNotPresent
-        env:
         {{- if eq .Values.aws.access "external" }}
+        env:
         - name: AWS_ACCESS_KEY_ID
           valueFrom:
             secretKeyRef:
-              name: {{ .Values.name }}-route53-credentials
+              name: {{ .Release.Name }}-route53-credentials
               key: aws_access_key_id
         - name: AWS_SECRET_ACCESS_KEY
           valueFrom:
             secretKeyRef:
-              name: {{ .Values.name }}-route53-credentials
+              name: {{ .Release.Name }}-route53-credentials
               key: aws_secret_access_key
         - name: AWS_DEFAULT_REGION
           value: {{ .Values.aws.region }}
         {{- end }}
         args:
-        {{- range .Values.sources }}
+        {{- range .Values.externalDNS.sources }}
         - --source={{ . }}
         {{- end }}
+        {{- if .Values.externalDNS.annotationFilter }}
+        - --annotation-filter={{ .Values.externalDNS.annotationFilter }}
+        {{- end }}
+        {{- if .Values.externalDNS.interval }}
+        - --interval={{ .Values.externalDNS.interval }}
+        {{- end }}
         - --provider={{ .Values.provider }}
         {{- if eq .Values.provider "azure" }}
         - --azure-config-file=/azure/config/azure.yaml
-        {{- end }}
-        {{- if eq .Values.aws.access "external" }}
-        - --domain-filter={{ .Values.aws.baseDomain }}
-        - --aws-zone-type=public
```

Comment:
we don't place the access type logic in the template now - the _helpers file handles this now.